cli/src/main/java/com/devonfw/tools/ide/io/AbstractIdeProgressBar.java

```diff
@@ -0,0 +1,68 @@
+package com.devonfw.tools.ide.io;
+
+/**
+ * Abstract implementation of {@link IdeProgressBar}.
+ */
+public abstract class AbstractIdeProgressBar implements IdeProgressBar {
+
+  private long currentProgress;
+  private final long maxLength;
+
+  /**
+   * @param maxLength the maximum length of the progress bar.
+   */
+  public AbstractIdeProgressBar(long maxLength) {
+
+    this.maxLength = maxLength;
+  }
+
+  /**
+   * Increases the progress bar by given step size.
+   *
+   * @param stepSize size to step by.
+   * @param currentProgress current progress state.
+   */
+  protected abstract void doStepBy(long stepSize, long currentProgress);
+
+  /**
+   * Increases the progress bar by given step size.
+   *
+   * @param stepSize size to step by.
+   */
+  protected abstract void doStepBy(long stepSize);
```

Comment:
IMHO we can now remove this method or why do we have both `doStepBy` methods?
```suggestion
```

Reply:
I've removed the method now.